### DISCUSSION TITLE
Exceptions aren't shown in Uvicorn log anymore (since 0.108)

### DISCUSSION THREAD

**Discussion by estyrke** [fastapi/fastapi#11060] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn # 0.25.0  
from fastapi import FastAPI, Request # 0.109.0 
from fastapi_versioning import VersionedFastAPI, version # 0.10.0 

app = FastAPI()


@app.get("/")
@version(1, 0)
def route():
    raise TypeError("Foo")


app = VersionedFastAPI(app)


@app.middleware("http")
async def session_middleware(
    request: Request,
    call_next,
):
    response = await call_next(request)
    return response


uvicorn.run(
    app,
    host="127.0.0.1",
)
```

### Description

* Trigger with `curl http://localhost:8000/v1_0/`

Returns "Internal Server Error". The Uvicorn log shows just "INFO:     127.0.0.1:54424 - "GET /v1_0/ HTTP/1.1" 500 Internal Server Error". With fastapi 0.107 and earlier the full traceback is written in the log:

```
INFO:     127.0.0.1:55224 - "GET / HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/Users/emil/Code/arbo-api/.venv/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[...]
  File "/Users/emil/Code/arbo-api/test-logging.py", line 11, in route
    raise TypeError("Foo")
TypeError: Foo
```

Apparently, removing either VersionedFastAPI or the middleware gets the exception back in the logs, so there's something in the intersection between those that causes the issue. I notice that the call_next in the middleware returns a 500 response, while in 0.107 it never returns but throws instead. It is almost impossible to develop when the strack traces aren't logged for programming errors. Can I work around this in some way? The real middleware is used to authenticate the user and update token cookies in the response, I am open to doing that with some other mechanism if possible.

### Operating System

macOS

### FastAPI Version

0.109.0

### Pydantic Version

2.5.3

### Python Version

3.11.6

**Comment by Kludex** [comment]
FastAPI Versioning is doing something. I can't reproduce it without it.

**Reply by estyrke** [reply]
Yes, I already noticed that, but I need versioning in the API. Is there some other way besides versioning_fastapi to accomplish that? I am asking for help resolving an issue here, not saying that this is necessarily FastAPI:s fault.

**Reply by Kludex** [reply]
I organize routers with the version, e.g. `APIRouter(prefix='/v1')`, and then I include all the v1 routers on it.

**Comment by estyrke** [comment, marked as answer]
I found a somewhat straightforward solution: I rewrote our session middleware as a "pure" ASGI middleware (as described in https://www.starlette.io/middleware/#pure-asgi-middleware) instead of using `@app.middleware`. It uglifies the code a bit, but at least it works. :)
